src/main/java/de/tum/in/www1/artemis/web/rest/NotificationSettingsResource.java

```diff
@@ -0,0 +1,94 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.badRequest;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.NotificationOption;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.NotificationOptionRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.NotificationSettingsService;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * REST controller for managing NotificationSettings (NotificationOptions).
+ */
+@RestController
+@RequestMapping("api/")
+public class NotificationSettingsResource {
+
+    private final Logger log = LoggerFactory.getLogger(NotificationSettingsResource.class);
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final NotificationOptionRepository notificationOptionRepository;
+
+    private final UserRepository userRepository;
+
+    private final NotificationSettingsService notificationSettingsService;
+
+    public NotificationSettingsResource(NotificationOptionRepository notificationOptionRepository, UserRepository userRepository,
+            NotificationSettingsService notificationSettingsService) {
+        this.notificationOptionRepository = notificationOptionRepository;
+        this.userRepository = userRepository;
+        this.notificationSettingsService = notificationSettingsService;
+    }
+
+    /**
+     * GET notification-settings/fetch-options : Get all NotificationOptions for current user
+     *
+     * Fetches the NotificationOptions for the current user from the server.
+     * These are only the options that the user has already modified.
+     * NotificationOptions (Server) corresponds to NotificationOptionCores (Client)#
+     *
+     * @return the list of found NotificationOptions
+     */
+    @GetMapping("notification-settings/fetch-options")
```

Comment:
This is not the correct REST URL, see the guidelines.

Reply:
```suggestion
    @GetMapping("notification-settings")
```

Reply:
I applied the change.